### PR TITLE
Feature/ins 184 contacts tableview

### PR DESCRIPTION
**Pull request by vierrra**

### CODE REVIEW

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 1]
```diff
@@ -35,85 +32,92 @@ public class ContactsCard: UITableViewCell {
         photoImageView = nil
     }
     
-    public func configureViewFor(construction: ContactsInformation) {
+    public func configureViewFor(contacts: ContactsInformation) {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Change the parameter name to contact. It's a single element, so it should be singular.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 2]
```diff
@@ -35,85 +32,92 @@ public class ContactsCard: UITableViewCell {
...
-            self.addSubview(callButton)
+        if let callButton = self.phoneButton {
+           self.addSubview(callButton)
 
-            self.callButton?.anchor(top:      self.topAnchor,
-                                    bottom:   self.bottomAnchor,
-                                    trailing: self.trailingAnchor,
-                                    padding:  UIEdgeInsets(top: 18, left: 0, bottom: 18, right: 16),
-                                    size:     CGSize(width: 20, height: 20))
+           self.phoneButton?.anchor(top:      self.topAnchor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift:106]: Indent this block.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 3]
```diff
@@ -134,11 +138,11 @@ public class ContactsCard: UITableViewCell {
         mainView.layer.applySketchShadow(color: .lightGray, alpha: 0.2, x: 16, y: 1, blur: 0, spread: 0)
     }
     
-    fileprivate func configure(name: String, office: String) {
-        nameLabel   = UILabel(frame: CGRect(x: 68, y: 173, width: 210, height: 19))
-        officeLabel = UILabel(frame: CGRect(x: 68, y: 194, width: 210, height: 14))
+    fileprivate func configure(name: String, role: String, mail: String, phone: String) {
+        nameLabel   = UILabel(frame: CGRect(x: 68, y: 11, width: 210, height: 19))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 4]
```diff
@@ -12,13 +12,15 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
...
+    public var presenterDelegate: UIViewController?
     weak var tableViewDelegate:   ContactsTableViewDidSelectDelegate?
     
     var searchBarIcon        = UIImage()
     var searchBarPlaceholder = String()
     var searchOnTableView    = String()
     var emptyMessage         = String()
-    public var constructionSiteList = [ContactsInformation]()
+    public var contactsSiteList = [ContactsInformation]()
+
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:23]: Remove extra line.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 5]
```diff
@@ -30,14 +32,14 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
            super.init(frame: frame)
     }
     
-    public func configureViewWith(constructionList:     [ContactsInformation],
+    public func configureViewWith(contactsList:     [ContactsInformation],
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 6]
```diff
@@ -71,10 +73,11 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
         
         tableView?.register(ContactsCard.self, forCellReuseIdentifier: ContactsCard.identifier())
         
-        tableView?.configureTableViewWith(constructionList:   constructionSiteList,
+        tableView?.configureTableViewWith(contactsList:   contactsSiteList,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:75]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 7]
```diff
@@ -71,10 +73,11 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
...
                                           animationDelegate:  self,
                                           selectCellDelegate: tableViewDelegate)
         tableView?.backgroundColor = .clear
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:78]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 8]
```diff
@@ -120,16 +123,16 @@ public class ContactsTableViewWithCollapseSearchBar: UIView {
         emptyMessageLabel.numberOfLines = 0
     }
     
-    public func updateConstructionSiteList(_ constructionSites: [ContactsInformation]) {
-        self.constructionSiteList                    = constructionSites
-        self.tableView?.filteredConstructionSiteList = constructionSites
+    public func updateContactsList(_ contacts: [ContactsInformation]) {
+        self.contactsSiteList                    = contacts
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:126]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 9]
```diff
@@ -11,14 +11,16 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
...
+    public var filteredContactsList   = [ContactsInformation]()
     public var currentHeightConstant: CGFloat = DefaultSearchBar.searchViewHeight
     public let maxHeaderHeight: CGFloat       = DefaultSearchBar.searchViewHeight
     public let minHeaderHeight: CGFloat       = 0
     
-    public var animationDelegate: ContactsAndHideSearchTableViewDelegate?
-    var selectCellDelegate:       ContactsTableViewDidSelectDelegate?
+    public var animationDelegate:  ContactsAndHideSearchTableViewDelegate?
+    public var selectCellDelegate: ContactsTableViewDidSelectDelegate?
+
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift:22]: Remove extra line.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 10]
```diff
@@ -28,22 +30,22 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
         super.init(frame: frame, style: style)
     }
     
-    public func configureTableViewWith(constructionList:   [ContactsInformation],
+    public func configureTableViewWith(contactsList:   [ContactsInformation],
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift:33]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 11]
```diff
@@ -54,17 +56,19 @@ public class ContactsTableView: UITableView, UITableViewDelegate, UITableViewDat
         guard let cell = tableView.dequeueReusableCell(withIdentifier: ContactsCard.identifier(), for: indexPath) as? ContactsCard else {
             return UITableViewCell()
         }
-        
-        cell.configureViewFor(construction: filteredConstructionSiteList[indexPath.row])
 
+
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Remove extra line.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 12]
```diff
@@ -9,66 +9,37 @@
...
-     let office                       = "Developer IOS "
-     let cardUrl                      = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT0P8RsPCO0qj21UeIfaVkRcsqguonI6bP4iLr3tWwS4qIS4MSquw"
-     let iconSize: CGSize             = CGSize(width: 35, height: 35)
-     let validMailButtons             = [UIButton(), nil, nil, UIButton(), nil, UIButton()]
-     let validPhoneButtons            = [UIButton(), UIButton(), nil, UIButton(), UIButton(), nil]
+class ConstructionContactsViewController: UIViewController, ContactsTableViewShowPresenter {
+    let navigationBarHeight: CGFloat = 60
+    let name                         = "Steve Jobs"
+    let role                         = "Developer IOS"
+    let contactPhoto                = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT0P8RsPCO0qj21UeIfaVkRcsqguonI6bP4iLr3tWwS4qIS4MSquw"
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 13]
```diff
@@ -9,66 +9,37 @@
...
+    let contactEmail                 = "stant@stant.com.br"
+    let contactPhone                 = "99999-9999"
     
-     var constructionSiteList         = [ContactsInformation]()
+    var contactsSiteList             = [ContactsInformation]()
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Remove extra spaces.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 14]
```diff
@@ -9,66 +9,37 @@
...
-                                                            mailImage:  UIImage(named: "mail") ?? UIImage(),
-                                                            callButton: self.validPhoneButtons[i],
-                                                            callImage:  UIImage(named: "call") ?? UIImage()))
+        for i in 0...4 {
             
-            self.setButtonsActionsFor(index: i)
+            contactsSiteList.append(ContactsInformation(name: name + "\(i)", role: role, photo: contactPhoto, mail: (i > 2 ? contactEmail : ""), phone: (i < 2 ? contactPhone : "")))
         }
         
         let contactsTableView = ContactsTableViewWithCollapseSearchBar(frame: CGRect(x: 0,
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift:40]: Indent the initialisation parameters.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 15]
```diff
@@ -78,18 +49,19 @@ class ConstructionContactsViewController: UIViewController {
                                  bottom:   0,
                                  right:    0))
         
-        contactsTableView.configureViewWith(constructionList: constructionSiteList, searchBarIcon: UIImage(named: "search") ?? UIImage(),
+        contactsTableView.configureViewWith(contactsList: contactsSiteList, searchBarIcon: UIImage(named: "search") ?? UIImage(),
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Indente parameters.

**Review by Leandro97 — CHANGES_REQUESTED** [review]

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 1 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 2 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift:106]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 3 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 4 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:23]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 5 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 11 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 12 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 13 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 14 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift:40]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 15 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

**Review by Leandro97 — APPROVED** [review]

**Review by andersonfernandes — APPROVED** [review]

**Pull request merged**